pkg/executor/push.go

```diff
@@ -58,6 +58,13 @@ func CheckPushPermissions(opts *config.KanikoOptions) error {
 	checked := map[string]bool{}
 	for _, destination := range opts.Destinations {
 		destRef, err := name.NewTag(destination, name.WeakValidation)
+		if opts.Insecure {
+			newReg, err := name.NewRegistry(destRef.RegistryStr(), name.WeakValidation)
```

Comment:
I think you want to pass in the insecure option to `name.NewRegistry` -- found [here](https://github.com/google/go-containerregistry/blob/master//pkg/name/options.go#L47) in the go-containerregistry code